Pattern: Use of non-ASCII character

Issue: -

## Description

Prevents non-ASCII characters.

Examples of **incorrect** code for this rule:

```js
console.log('Ｙｏ');
```

Examples of **correct** code for this rule:

```js
console.log('Yo');
```