Pattern: Malformed comment spacing

Issue: -

## Description

Prefer at least one space after slashes for comments.

Examples of **correct** code:

```swift
// This is a comment

/// Triple slash comment

// Multiline double-slash
// comment

/// Multiline triple-slash
/// comment

/// Multiline triple-slash
///   - This is indented

// - MARK: Mark comment

/* Asterisk comment */

/*
    Multiline asterisk comment
*/
   }
}
```

Examples of **incorrect** code:

```swift
//â†“Something

//â†“MARK

//â†“ğŸ‘¨â€ğŸ‘¨â€ğŸ‘¦â€ğŸ‘¦Something

func a() {
    //â†“This needs refactoring
    print("Something")
}
//â†“We should improve above function

///â†“This is a comment

/// Multiline triple-slash
///â†“This line is incorrect, though

//â†“- MARK: Mark comment

```

## Further Reading

* [SwiftLint - Test case accessibility](https://realm.github.io/SwiftLint/comment_spacing.html)