Pattern: Use of force unwrapping

Issue: -

## Description

Force unwrapping should be avoided.

Examples of **correct** code:
```swift
if let url = NSURL(string: query)


navigationController?.pushViewController(viewController, animated: true)


let s as! Test


try! canThrowErrors()


let object: Any!


@IBOutlet var constraints: [NSLayoutConstraint]!


setEditing(!editing, animated: true)


navigationController.setNavigationBarHidden(!navigationController.navigationBarHidden, animated: true)


if addedToPlaylist && (!self.selectedFilters.isEmpty || self.searchBar?.text?.isEmpty == false) {}


print("\(xVar)!")


var test = (!bar)


var a: [Int]!


private var myProperty: (Void -> Void)!


func foo(_ options: [AnyHashable: Any]!) {

```
Examples of **incorrect** code:
```swift

let url = NSURL(string: query)â†“!


navigationControllerâ†“!.pushViewController(viewController, animated: true)


let unwrapped = optionalâ†“!


return cellâ†“!


let url = NSURL(string: "http://www.google.com")â†“!


let dict = ["Boooo": "ðŸ‘»"]func bla() -> String { return dict["Boooo"]â†“! }


let dict = ["Boooo": "ðŸ‘»"]func bla() -> String { return dict["Boooo"]â†“!.contains("B") }


let a = dict["abc"]â†“!.contains("B")


dict["abc"]â†“!.bar("B")


if dict["a"]â†“!!!! {


var foo: [Bool]! = dict["abc"]â†“!


context("abc") {
  var foo: [Bool]! = dict["abc"]â†“!
}


open var computed: String { return foo.barâ†“! }

```

## Further Reading

* [SwiftLint - Force Unwrapping](https://realm.github.io/SwiftLint/force_unwrapping.html)