Pattern: Leading whitespace

Issue: -

## Description

Files should not contain leading whitespace.

Examples of **correct** code:
```swift
//

```
Examples of **incorrect** code:
```swift


 //

```

## Further Reading

* [SwiftLint - Leading Whitespace](https://realm.github.io/SwiftLint/leading_whitespace.html)