Pattern: Malformed trailing `,`

Issue: -

## Description

Trailing commas in arrays and dictionaries should be avoided or enforced.

Examples of **correct** code:
```swift
let foo = [1, 2, 3]


let foo = []


let foo = [:]


let foo = [1: 2, 2: 3]


let foo = [Void]()


let example = [ 1,
 2
 // 3,
]


foo([1: "\(error)"])

```
Examples of **incorrect** code:
```swift

let foo = [1, 2, 3â†“,]


let foo = [1, 2, 3â†“, ]


let foo = [1, 2, 3   â†“,]


let foo = [1: 2, 2: 3â†“, ]


struct Bar {
 let foo = [1: 2, 2: 3â†“, ]
}


let foo = [1, 2, 3â†“,] + [4, 5, 6â†“,]


let example = [ 1,
2â†“,
 // 3,
]


let foo = ["××‘×’", "Î±Î²Î³", "ğŸ‡ºğŸ‡¸"â†“,]

```

## Further Reading

* [SwiftLint - Trailing Comma](https://github.com/realm/SwiftLint/blob/master/Rules.md#trailing-comma)