Pattern: Trailing newline

Issue: -

## Description

Files should have a single trailing newline.

Examples of **correct** code:
```swift
let a = 0

```
Examples of **incorrect** code:
```swift

let a = 0


let a = 0

```

## Further Reading

* [SwiftLint - Trailing Newline](https://realm.github.io/SwiftLint/trailing_newline.html)